panels/examples/complexbin/animal.go

```diff
@@ -0,0 +1,110 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/vimeo/dials/panels"
+)
+
+var _ = panels.Must(p.Register("animals", defaultAnimals()))
+
+type animal struct {
+	Type  string // dog, lion, duck
+	Legs  int8
+	Group string // mammal, reptile, marsupial
+}
+
+type animals struct {
+	m map[string]animal
+}
+
+type animalTraits struct {
+	Name string
+}
+
+func (a *animals) Description(scPath []string) string {
+	return "Get a list of all the animals in the directory that match the criteria"
+}
+
+func (a *animals) ShortUsage(scPath []string) string {
+	return strings.Join(scPath, " ")
+}
+
+func (a *animals) LongUsage(scPath []string) string {
+	return strings.Join(scPath, " ") + "\n\tget all the animals"
+}
+
+func (a *animals) DefaultConfig() interface{}      { return &animalTraits{} }
+func (a *animals) SetupParams() panels.SetupParams { return panels.SetupParams{} }
+
+func (a *animals) Run(ctx context.Context, s *panels.Something) error {
+	var sb strings.Builder
```

Comment:
I don't think there's any benefit from using `strings.Builder` in this function.
We might as well write directly to `s.W`.